### PR TITLE
Support stream option for Azure Blob storage

### PR DESCRIPTION
**Pull request by dwaijam**
Provided an implementation of the Open method of AzureCloudBlockBlob and for the Open method of
AzureBlockBlobFileSystem which was previously a not supported one.

### CODE REVIEW

File: lang/cs/Org.Apache.REEF.IO.Tests/TestAzureBlockBlobFileSystem.cs [hunk 1]
```diff
@@ -45,9 +45,11 @@ public void TestCreateNotSupported()
         }
 
         [Fact]
-        public void TestOpenNotSupported()
+        public void TestOpen()
         {
-            Assert.Throws<NotSupportedException>(() => new TestContext().GetAzureFileSystem().Open(FakeUri));
+            var testContext = new TestContext();
+            testContext.GetAzureFileSystem().Open(new Uri(FakeUri, "container/file"));
+            testContext.TestCloudBlockBlob.Received(1).Open();
```
Comment (ghost) [inline lang/cs/Org.Apache.REEF.IO.Tests/TestAzureBlockBlobFileSystem.cs]: Make sure to add assert logic to the Open result.

Comment (dwaijam) [reply lang/cs/Org.Apache.REEF.IO.Tests/TestAzureBlockBlobFileSystem.cs]: Nothing to Assert on the Open result as GetAzureFileSystem() returns a proxy which return null on the Open method call

Comment (ghost) [reply lang/cs/Org.Apache.REEF.IO.Tests/TestAzureBlockBlobFileSystem.cs]: Generally you should always test what is returned from the function you are testing. This test only checks that an internal request to TestCloudBlockBlob is called. You aren't actually testing if the stream opened by TestCloudBlockBlob is properly returned. Maybe the implementation of IFileSystem.Open always returns null? The test should valid that this is not the case.

File: lang/cs/Org.Apache.REEF.IO.Tests/TestAzureBlockBlobFileSystem.cs [hunk 2]
```diff
@@ -144,8 +148,10 @@ public IFileSystem GetAzureFileSystem()
                 injector.BindVolatileInstance(TestCloudBlobClient);
                 var fs = injector.GetInstance<AzureBlockBlobFileSystem>();
                 TestCloudBlobClient.BaseUri.ReturnsForAnyArgs(FakeUri);
+                TestCloudBlockBlob.Open().Returns(TestStream);
```
Comment (ghost) [inline lang/cs/Org.Apache.REEF.IO.Tests/TestAzureBlockBlobFileSystem.cs:151]: I would put this logic in the TestOpen() function, but this is what I wanted. Thank you for adding it.

**Review by ghost — APPROVED** [review]

**Pull request merged**